### ISSUE TITLE
DPL-1100 - As developers we need to fix and merge the dependencies to reduce security problems and reduce techncial debt

### ISSUE THREAD

**Issue by stevieing** [sanger/crawler#909]
**Describe the Housekeeping**
As developers we need to fix and merge the dependencies to reduce security problems and reduce techncial debt

**Blocking issues**
Describe any other issues or tickets that may be blocking this change.

**Additional context**

**Comment by stevieing** [comment]
List of dependencies to update. Please ensure to do a rebase before commiting as they may be out of date:
- #890
- #897
- #899
- #901
- #902
- #906
- #907
- #908

**Comment by dasunpubudumal** [comment]
Attaching some finds on #890:

The error message from [CI log](https://github.com/sanger/crawler/actions/runs/7876865812/job/21761508956) is attached below:

```
=================================== FAILURES ===================================
___________________ test_run_mysql_executemany_query_success ___________________
config = <module 'crawler.config.test' from '/home/runner/work/crawler/crawler/crawler/config/test.py'>
    def test_run_mysql_executemany_query_success(config):
>       conn = MockMySQLConnection()
E       TypeError: Can't instantiate abstract class MockMySQLConnection with abstract methods cmd_change_user, cmd_debug, cmd_init_db, cmd_ping, cmd_process_kill, cmd_query, cmd_query_iter, cmd_quit, cmd_refresh, cmd_reset_connection, cmd_shutdown, cmd_statistics, cmd_stmt_close, cmd_stmt_execute, cmd_stmt_prepare, cmd_stmt_reset, cmd_stmt_send_long_data, connection_id, get_row
tests/db/test_mysql.py:55: TypeError
```

The update from `8.2.0` to `8.3.0` adds an additional decorator `@abstractmethod` (coming from `abc` module (i.e., **A**bstract **B**ase **C**lasses)) for the abstract methods in the superclass `MySQLConnectionAbstract`, that we use to create a `MockMySQLConnection` via inheritance. I didn't see this change in their [changelog](https://dev.mysql.com/doc/relnotes/connector-python/en/news-8-3-0.html), but was able to find out by comparing the source code versions.

<img width="1912" alt="image" src="https://github.com/sanger/crawler/assets/25484917/be7ac439-348f-4757-82a7-fcb5a3bff16c">

If a subclass that does not implement all the methods decorated with `@abstractmethod` is instantiated, Python interpreter will throw out a [`TypeError`](https://ioflood.com/blog/python-abstract-class/#:~:text=If%20a%20subclass%20doesn't,an%20instance%20of%20the%20subclass.&text=The%20solution%20is%20to%20ensure,abstract%20methods%20of%20the%20superclass.), as it was [designed to do so](https://docs.python.org/3/library/abc.html#abc.abstractmethod).

I do not think that this will affect our database connection initiation logic in our business logic, as it uses the helper function `mysql.connect()` that returns a `MySQLConnection` object (which implements the abstract methods), but it will require code changes in **tests** when we're trying to initiate a mock connection.

**Comment by dasunpubudumal** [comment]
Furthermore, the upgrade from `8.2.0` to `8.3.0` carries significant typing changes. One (out of many) such example is shown below.

<img width="1912" alt="Screenshot 2024-02-20 at 5 25 34 pm" src="https://github.com/sanger/crawler/assets/25484917/aea9c568-922d-477f-9522-754a78be77fd">

We might have to update our function signatures and function calls to adhere to the upgrades; otherwise, `mypy` will complain regarding typing information.

**Comment by dasunpubudumal** [comment]
The changes discussed in https://github.com/sanger/crawler/issues/909#issuecomment-1953959198 and https://github.com/sanger/crawler/issues/909#issuecomment-1954699740 were completed and merged into develop (and master) with `v2.13.0` release.
